lib/bucket_store/in_memory.rb

```diff
@@ -24,21 +24,17 @@ def reset!
       @buckets = Hash.new { |hash, key| hash[key] = {} }
     end
 
-    def upload!(bucket:, key:, content:)
-      @buckets[bucket][key] = content
+    def upload!(bucket:, key:, file:)
+      @buckets[bucket][key] = file.read
 
       {
         bucket: bucket,
         key: key,
       }
     end
 
-    def download(bucket:, key:)
-      {
-        bucket: bucket,
-        key: key,
-        content: @buckets[bucket].fetch(key),
-      }
+    def download(bucket:, key:, file:)
+      file.write(@buckets[bucket].fetch(key))
```

Comment:
Think we're missing a `.rewind` here, as otherwise we can't read the buffer back straight away which is what gcs/s3 allow (similar thing for the disk adapter)
```ruby
[1] pry(main)> require "bucket_store"
=> true
[2] pry(main)> BucketStore.for("inmemory://bucket/test.txt").upload!("hello")
I, [2023-04-20T09:57:16.487789 #78930]  INFO -- : {:event=>"key_storage.upload_started", :bucket=>"bucket", :key=>"test.txt", :adapter_type=>:inmemory}
I, [2023-04-20T09:57:16.487939 #78930]  INFO -- : {:event=>"key_storage.upload_finished", :duration=>1.4999997802078724e-05, :bucket=>"bucket", :key=>"test.txt", :adapter_type=>:inmemory}
=> "inmemory://bucket/test.txt"
[3] pry(main)> s = StringIO.new
=> #<StringIO:0x000000011859e2f0>
[4] pry(main)> BucketStore.for("inmemory://bucket/test.txt").stream.download(file: s)
I, [2023-04-20T09:57:26.061935 #78930]  INFO -- : {:event=>"key_storage.download_started"}
I, [2023-04-20T09:57:26.061979 #78930]  INFO -- : {:event=>"key_storage.download_finished", :duration=>8.000002708286047e-06}
=> {:bucket=>"bucket", :key=>"test.txt", :file=>#<StringIO:0x000000011859e2f0>}
[5] pry(main)> s.read
=> ""
[6] pry(main)> s.rewind
=> 0
[7] pry(main)> s.read
=> "hello"
```

Reply:
@piiraa  - think this still needs doing as well?